<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Controller Commands</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label_command">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>20</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Command:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_select_motor">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>60</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Select Motor:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="combo_motor">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>60</y>
      <width>271</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_arg1">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>100</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Arg1:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="line_arg1">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>100</y>
      <width>271</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_arg2">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>140</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Arg2:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="line_arg2">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>140</y>
      <width>271</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="button_send">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>180</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Send Command</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_motor_info">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>230</y>
      <width>351</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Motor Status and Position</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="text_motor_status">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>260</y>
      <width>581</width>
      <height>120</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
    <property name="placeholderText">
     <string>Motor status will be displayed here...</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_target_position">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>400</y>
      <width>351</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Set Target Position for Motor:</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spinbox_position">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>430</y>
      <width>120</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <double>0.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QTextEdit" name="text_output">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>470</y>
      <width>581</width>
      <height>221</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
    <property name="placeholderText">
     <string>Command output will be displayed here...</string>
    </property>
   </widget>
   <widget class="QComboBox" name="combo_command">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>20</y>
      <width>271</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>combo_motor</tabstop>
  <tabstop>line_arg1</tabstop>
  <tabstop>line_arg2</tabstop>
  <tabstop>button_send</tabstop>
  <tabstop>text_motor_status</tabstop>
  <tabstop>spinbox_position</tabstop>
  <tabstop>text_output</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
